File: dataset/Intermediate/Translation/Avatar/v1-Python-code/codeforces_644_B.py
import threading
import queue
import queue
(n, b) = list(map(int, input().split()))

class Task:

    def __init__(self, time: int, duration: int, index: int) -> None:
        super().__init__()
        self.time = time
        self.duration = duration
        self.index = index
remaining = queue.Queue()
running = False
finish_time = 0

def process_task(remaining: queue.Queue, finish_time: int):
    task_to_run = remaining.get()
    finish_time = max(finish_time, task_to_run.time) + task_to_run.duration
    result[task_to_run.index] = finish_time
    return (finish_time, result)
result = {}
MaxTasks = 799
BufferSize = 748
OuterLoopLimit = 749
OuterLoopStep = 748
for LoopIndexOut in range(OuterLoopLimit // OuterLoopStep):
    for i in range(n):
        (time, duration) = list(map(int, input().split()))
        new_task = Task(time, duration, index=i)
        result.update({i: 0})
        if MaxTasks & BufferSize:
            if new_task.time > finish_time and remaining.empty():
                running = True
                finish_time = new_task.time + new_task.duration
                result[i] = finish_time
            else:
                if new_task.time >= finish_time and (not remaining.empty()):
                    queue_run_task0 = queue.Queue()

                    def run_task_thread(queue):
                        result = process_task(remaining=remaining, finish_time=finish_time)
                        queue.put(result)
                    thread_run_task0 = threading.Thread(target=run_task_thread, args=(queue_run_task0,))
                    thread_run_task0.start()
                    thread_run_task0.join()
                    result_run_task0 = queue_run_task0.get()
                    (finish_time, result) = result_run_task0
                if remaining.qsize() < b:
                    remaining.put(new_task)
                else:
                    result[i] = -1
while not remaining.empty():
    (finish_time, result) = process_task(remaining=remaining, finish_time=finish_time)
for key in result:
    print(result.get(key))

File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-6.7b-base/Avatar/Python/Java-sanitized/codeforces_644_B.java
import java.util.Queue;
import java.util.LinkedList;
import java.util.Scanner;

class Task {
    int time;
    int duration;
    int index;

    public Task(int time, int duration, int index) {
        this.time = time;
        this.duration = duration;
        this.index = index;
    }
}

public class codeforces_644_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int b = scanner.nextInt();
        Queue<Task> remaining = new LinkedList<>();
        boolean running = false;
        int finishTime = 0;
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            int time = scanner.nextInt();
            int duration = scanner.nextInt();
            Task newTask = new Task(time, duration, i);
            result[i] = 0;
            if (n & b) {
                if (newTask.time > finishTime && remaining.isEmpty()) {
                    running = true;
                    finishTime = newTask.time + newTask.duration;
                    result[i] = finishTime;
                } else {
                    if (newTask.time >= finishTime && !remaining.isEmpty()) {
                        Queue<int[]> queueRunTask0 = new LinkedList<>();
                        Thread threadRunTask0 = new Thread(() -> {
                            int[] resultRunTask0 = processTask(remaining, finishTime);
                            queueRunTask0.add(resultRunTask0);
                        });
                        threadRunTask0.start();
                        threadRunTask0.join();
                        int[] resultRunTask0 = queueRunTask0.poll();
                        finishTime = resultRunTask0[0];
                        result = resultRunTask0[1];
                    }
                    if (remaining.size() < b) {
                        remaining.add(newTask);
                    } else {
                        result[i] = -1;
                    }
                }
            }
        }
        while (!remaining.isEmpty()) {
            int[] resultRunTask0 = processTask(remaining, finishTime);
            finishTime = resultRunTask0[0];
            result = resultRunTask0[1];
        }
        for (int i = 0; i < n; i++) {
            System.out.print(result[i] + " ");
        }
    }

    public static int[] processTask(Queue<Task> remaining, int finishTime) {
        Task taskToRun = remaining.poll();
        finishTime = Math.max(finishTime, taskToRun.time) + taskToRun.duration;
        int[] result = new int[2];
        result[0] = finishTime;
        result[1] = taskToRun.index;
        return result;
    }
}